content-security-policy/base-uri/base-uri_iframe_sandbox.sub.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>base-uri works correctly inside a sandboxed iframe.</title>
+    <script src='/resources/testharness.js'></script>
+    <script src='/resources/testharnessreport.js'></script>
+
+    <!-- CSP served: base-uri 'self' -->
+</head>
+
+<body>
+    <h1>base-uri works correctly inside a sandboxed iframe.</h1>
+    <div id='log'></div>
+
+    <script>
+        window.addEventListener('securitypolicyviolation', function(e) {
+            assert_unreached('No CSP violation report has fired.');
+        });
+
+        async_test(function(t) {
+            window.addEventListener('message', t.step_func(function(e) {
+                if (e.data.startsWith('SAME-ORIGIN,')) {
```

Comment:
I'd prefer for you to validate the source of the message. That is, rename the iframe element variable on line 29 to something like `i`, then compare `e.source == i.contentWindow`.